pkg/operator/csi/csidrivernodeservicecontroller/csi_driver_node_service_controller.go

```diff
@@ -114,27 +118,41 @@ func (c *CSIDriverNodeServiceController) Name() string {
 
 func (c *CSIDriverNodeServiceController) sync(ctx context.Context, syncContext factory.SyncContext) error {
 	opSpec, opStatus, _, err := c.operatorClient.GetOperatorState()
+	if management.IsOperatorRemovable() && errors.IsNotFound(err) {
+		return nil
+	}
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil
-		}
 		return err
 	}
 
 	if opSpec.ManagementState != opv1.Managed {
 		return nil
 	}
 
-	manifest := replacePlaceholders(c.manifest, opSpec)
-	required := resourceread.ReadDaemonSetV1OrDie(manifest)
+	meta, err := c.operatorClient.GetObjectMeta()
+	if err != nil {
+		return err
+	}
+	if management.IsOperatorRemovable() && meta.DeletionTimestamp != nil {
+		return c.syncDeleting(ctx, opSpec, opStatus, syncContext)
+	}
+	return c.syncManaged(ctx, opSpec, opStatus, syncContext)
+}
 
-	for i := range c.optionalDaemonSetHooks {
-		err := c.optionalDaemonSetHooks[i](opSpec, required)
+func (c *CSIDriverNodeServiceController) syncManaged(ctx context.Context, opSpec *opv1.OperatorSpec, opStatus *opv1.OperatorStatus, syncContext factory.SyncContext) error {
+	klog.V(4).Infof("syncManaged")
+	if management.IsOperatorRemovable() {
+		err := c.operatorClient.EnsureFinalizer(c.name)
```

Comment:
nice.  We're allowed to have something.openshift.io/something, right? Will this be easier as `removal.operator.openshift.io/name`?  that fans out cleanly and doesn't burn a "good" name.

Reply:
do we need to encode an overall operator name in there too?  We haven't made that concept so far, but we could to distinguish which operator needs to act.

Reply:
added `<operator name>.operator.openshift.io/<controller name>`. Where `operator_name` is either `os.Args[0]` or `$OPERATOR_NAME`.